frontend/src/ui/components/PreviewDropdown/PreviewDropdown.tsx

```diff
@@ -0,0 +1,57 @@
+import * as React from 'react';
+import { Collapse, Nav } from 'react-bootstrap';
+import { WrappedMessage } from 'utils/intl';
+import messages from './displayMessages';
+import './styles.scss';
+
+interface PreviewDropdownProps {
+  children?: React.ReactNode;
+  initialExpanded?: boolean;
+  handleChange: Function;
+}
+
+export const PreviewDropdown: React.FC<PreviewDropdownProps> = (
+  props: PreviewDropdownProps
+) => {
+  const [open, setOpen] = React.useState(false);
+  const [selectedPreview, changePreview] = React.useState('dashboard');
+  const updatePreview = (key: string) => {
+    changePreview(key);
+    setOpen(!open);
+    props.handleChange(key);
+  };
+  const children = () => {
+    const toRender: JSX.Element[] = [];
+
+    Object.entries(messages).forEach(([key, _]) =>
+      toRender.push(
+        <button onClick={() => updatePreview(key)} key={key} type="button">
+          <p>
+            <WrappedMessage messages={messages} id={key} />
+          </p>
+        </button>
+      )
+    );
+
+    return <div>{toRender}</div>;
+  };
```

Comment:
This is a bit hacky and not easy to understand. Messages aren't something we should rely on to generate markup. 

Probably we can describe dropdown items as a plain JSX?